engine/apps/twilioapp/phone_provider.py

```diff
@@ -217,24 +219,47 @@ def _parse_number(self, number: str):
             return False, None, None
 
     @property
-    def _twilio_api_client(self):
+    def _default_twilio_api_client(self):
         if live_settings.TWILIO_API_KEY_SID and live_settings.TWILIO_API_KEY_SECRET:
             return Client(
                 live_settings.TWILIO_API_KEY_SID, live_settings.TWILIO_API_KEY_SECRET, live_settings.TWILIO_ACCOUNT_SID
             )
         else:
             return Client(live_settings.TWILIO_ACCOUNT_SID, live_settings.TWILIO_AUTH_TOKEN)
 
+    @property
+    def _default_twilio_number(self):
+        return live_settings.TWILIO_NUMBER
+
+    def _twilio_sender(self, sender_type, to):
+        _, _, country_code = self._parse_number(to)
+        TwilioSender = apps.get_model("twilioapp", sender_type)
+        senders = list(TwilioSender.objects.filter(Q(country_code=country_code) | Q(country_code__isnull=True)))
```

Comment:
maybe you can use [order_by + asc or desc](https://docs.djangoproject.com/en/4.2/ref/models/expressions/#django.db.models.Expression.asc) here? (instead of sorting in Python) and then get the `.first()` from the resulting queryset ?